File: app/src/main/java/com/example/android/popularmovies/networkUtils/ApiService.java
package com.example.android.popularmovies.networkUtils;

import com.example.android.popularmovies.model.Response;
import com.example.android.popularmovies.model.Review;
import com.example.android.popularmovies.model.Trailer;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {

    @GET("popular")
    public Single<Response> getPopularMovies(@Query("api_key") String apiKey, @Query("language") String language);

    @GET("top_rated")
    public Single<Response> getTopRatedMovies(@Query("api_key") String apiKey, @Query("language") String language);

    @GET("{id}/videos")
    Single<Trailer> getMovieTrailers(@Path("id") int id, @Query("api_key") String apiKey, @Query("language") String language);

    @GET("{id}/reviews")
    Single<Review> getMovieReviews(@Path("id") int id, @Query("api_key") String apiKey, @Query("language") String language);
}

File: app/src/main/java/com/example/android/popularmovies/networkUtils/ApiService.kt
package com.example.android.popularmovies.networkUtils

import com.example.android.popularmovies.model.Response
import com.example.android.popularmovies.model.Review
import com.example.android.popularmovies.model.Trailer

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface ApiService {

  @GET("popular")
  fun getPopularMovies(
    @Query("api_key") apiKey: String, @Query("language") language: String): Single<Response>

  @GET("top_rated")
  fun getTopRatedMovies(@Query("api_key") apiKey: String, @Query("language") language: String): Single<Response>

  @GET("{id}/videos")
  fun getMovieTrailers(@Path("id") id: Int, @Query("api_key") apiKey: String, @Query("language") language: String): Single<Trailer>

  @GET("{id}/reviews")
  fun getMovieReviews(@Path("id") id: Int, @Query("api_key") apiKey: String, @Query("language") language: String): Single<Review>
}
